### PR TITLE
[Ruby]: Ruby / Messages bump

### PR DESCRIPTION
**Pull request by luke-hill**
### 🤔 What's changed?

Bumped minimum messages / ruby ver

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)
- :bug: Bug fix (non-breaking change which fixes a defect)
- :zap: New feature (non-breaking change which adds new behaviour)
- :boom: Breaking change (incompatible changes to the API)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: ruby/cucumber-gherkin.gemspec [hunk 1]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
     'source_code_uri' => 'https://github.com/cucumber/gherkin/blob/main/ruby'
   }
 
-  s.add_runtime_dependency 'cucumber-messages', '>= 19.1.4', '< 22.1'
+  s.add_runtime_dependency 'cucumber-messages', '~> 22.0'
```
Comment (mpkorstanje) [inline ruby/cucumber-gherkin.gemspec:23]: Unless something materially changed about the messages, is there a reason not to increase the upper bound instead?

Comment (luke-hill) [reply ruby/cucumber-gherkin.gemspec:23]: As in permit 19.1 but increase the upper limit to say 23?

Currently the maximum version is 22.0 so any revision in the 22.x should be fine here. Just trying to reduce the support surface area. In theory we could keep this support area the same?

Comment (mpkorstanje) [reply ruby/cucumber-gherkin.gemspec:23]: Yes. Allowing a wider range allows us to make major releases of messages without also making major releases of Gherkin. 

This reduces the churn somewhat.

Comment (luke-hill) [reply ruby/cucumber-gherkin.gemspec:23]: I mean in theory nothing stops us from setting the upper limit to 30 by the same logic. Or having no upper limit.

Bundler flags things without an upper limit hence the twiddle key. In theory something could come in v23 which breaks things hence setting this value?

Comment (mpkorstanje) [reply ruby/cucumber-gherkin.gemspec:23]: Yeah I would think `'>= 19.1.4', '< 23'` should be good.

Comment (luke-hill) [reply ruby/cucumber-gherkin.gemspec:23]: I'll make that change now.

**Comment by luke-hill** [comment]
@mpkorstanje - Ideally we would add in another breaking change for ruby alongside this, to kill 2 birds with one stone before releasing. If you're happy to hold off a short amount of time

**Comment by luke-hill** [comment]
In other news @mpkorstanje this is one that's slipped through. As this should trigger all ruby tests. I'll have a look to see the changes and see if I can work it out

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,13 +8,15 @@ This document is formatted according to the principles of [Keep A CHANGELOG](htt
...
+- [Go, Ruby, Perl] Upgraded messages to v22
+- [.NET] Bump sdk to .net 8. Added .net 8 to test platforms
 - [Go] Improve performance - don't compile regex on matcher create
 - [Perl] Fix release packaging
 - [Perl] Include CHANGELOG.md in tarball
-- [Perl] Upgrade messages to v22
-- [Perl] Harmonized error reporting with mainstream implementations  errors are now converted to messages and reported in the message stream ([#31](https://github.com/cucumber/gherkin/issues/31))
+- [Perl] Harmonized error reporting with mainstream implementations;
+  errors are now converted to messages and reported in the message stream
+  ([#31](https://github.com/cucumber/gherkin/issues/31))
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply CHANGELOG.md:18]: I just picked the HEAD, so it must have been wrong there. More work I'll get on top of on monday

**Comment by mpkorstanje** [comment]
I've pushed some changes to minimize the diff.

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpkorstanje) [reply CHANGELOG.md]: I'm guessing that was your local head and behind with was in the repo.

**Pull request merged**